### PR TITLE
[receiver/haproxy] move haproxyreceiver to beta

### LINKED ISSUE #27254: Move haproxyreceiver to beta
### Component(s)

receiver/haproxy

### Describe the issue you're reporting

I propose to move the haproxyreceiver to beta stability. It has had a few reports and has no bugs related to its API. The metrics haven't changed in the last 6 months.

### PR DESCRIPTION
**Pull request by atoulme**
**Description:** 
Move the haproxyreceiver to beta stability

**Link to tracking Issue:**
Fixes #27254

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]
I think @MovieStoreGuy (the other code owner) should approve as well

**Review by MovieStoreGuy — APPROVED** [review]

**Pull request merged**
